File: tests/fixtures/rust-simple/src/utils/logger.rs
/// A simple logger for the payment system.
pub struct Logger {
    prefix: String,
}

impl Logger {
    /// Create a new logger with a prefix.
    pub fn new(prefix: &str) -> Self {
        Self {
            prefix: prefix.to_string(),
        }
    }

    /// Log an info message.
    pub fn info(&self, message: &str) {
        println!("[{}] INFO: {}", self.prefix, message);
    }

    /// Log a warning message.
    pub fn warn(&self, message: &str) {
        eprintln!("[{}] WARN: {}", self.prefix, message);
    }

    /// Log an error message.
    pub fn error(&self, message: &str) {
        eprintln!("[{}] ERROR: {}", self.prefix, message);
    }
}

/// Static logger instance name.
pub static DEFAULT_LOGGER_NAME: &str = "scope";

File: tests/fixtures/java-simple/src/main/java/com/example/utils/Logger.java
package com.example.utils;

/**
 * Simple logging utility.
 */
public class Logger {

    private final String prefix;

    public Logger(String prefix) {
        this.prefix = prefix;
    }

    public void info(String message) {
        System.out.println("[INFO] " + prefix + ": " + message);
    }

    public void error(String message) {
        System.out.println("[ERROR] " + prefix + ": " + message);
    }

    public static Logger create(String name) {
        return new Logger(name);
    }
}
